azure/src/test/java/com/microsoft/azure/management/TestLoadBalancer.java

```diff
@@ -523,8 +522,8 @@ public LoadBalancer createResource(LoadBalancers resources) throws Exception {
 
         @Override
         public LoadBalancer updateResource(LoadBalancer resource) throws Exception {
-            Map<String, PublicIpAddress> existingPips = ensurePIPs(pips);
-            PublicIpAddress pip = existingPips.get(1);
+            ensurePIPs(pips);
+            PublicIpAddress pip = pips.getByGroup(GROUP_NAME, PIP_NAMES[1]);
```

Comment:
Usually as a developer I expect that if I pass in a list of resource names for resources to be created we should get the created resources back in same order. Rather than getting the resources in the order in which those created we should put those in order the request to those came.